.github/workflows/ts-sdk.yml

```diff
@@ -56,4 +56,46 @@ jobs:
       - name: Install dependencies
         run: pnpm install --frozen-lockfile
       - name: Run e2e tests
-        run: pnpm dlx concurrently --kill-others --success command-1 'cargo run --bin sui-test-validator' 'pnpm sdk test:e2e'
+        run: pnpm dlx concurrently --kill-others --success command-1 'RUST_LOG="consensus=off" cargo run --bin sui-test-validator' 'pnpm sdk test:e2e'
+  end_to_end_devnet:
+    name: End-to-end tests against DevNet
+    needs: diff
+    if: needs.diff.outputs.isTypescriptSDK == 'true'
+    runs-on: [self-hosted, self-hosted-arc]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
+      - uses: pnpm/action-setup@v2.2.4
+        with:
+          version: 7
+      - run: cargo install --locked --git https://github.com/MystenLabs/sui.git --branch devnet sui
+      - name: Install Nodejs
+        uses: actions/setup-node@v3
+        with:
+          node-version: "16"
+          cache: "pnpm"
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile
+      - name: Run e2e tests
+        run: pnpm dlx concurrently --kill-others 'VITE_FAUCET_URL="https://faucet.devnet.sui.io:443/gas" VITE_FULLNODE_URL="https://fullnode.devnet.sui.io" pnpm sdk test:e2enw'
```

Comment:
Will we get rate limited on the faucet?

Reply:
no, this is running on self-hosted instance, which are whitelisted from the rate-limiting